internal/services/consumer.go

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright Strimzi authors.
+// License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+//
+
+// Package services defines an interface for canary services and related implementations
+package services
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/Shopify/sarama"
+	"github.com/strimzi/strimzi-canary/internal/config"
+)
+
+type Consumer struct {
+	config        *config.CanaryConfig
+	consumerGroup sarama.ConsumerGroup
+}
+
+func NewConsumer(config *config.CanaryConfig) Service {
+	// TODO: add specific consumer configuration
+	consumerGroup, err := sarama.NewConsumerGroup([]string{config.BootstrapServers}, config.ConsumerGroupID, nil)
+	if err != nil {
+		log.Printf("Error creating the Sarama consumer: %v", err)
+		panic(err)
+	}
+	c := Consumer{
+		config:        config,
+		consumerGroup: consumerGroup,
+	}
+	return &c
+}
+
+func (c *Consumer) Start() {
+	cgh := &consumerGroupHandler{}
+	ctx := context.Background()
+	go func() {
+		for {
+			// this method calls the methods handler on each stage: setup, consume and cleanup
+			c.consumerGroup.Consume(ctx, []string{c.config.Topic}, cgh)
+		}
+	}()
+}
+
+func (c *Consumer) Stop() {
+	log.Printf("Stopping consumer")
+	err := c.consumerGroup.Close()
+	if err != nil {
+		log.Printf("Error closing the Sarama consumer: %v", err)
+		os.Exit(1)
+	}
+	log.Printf("Consumer closed")
+}
+
+// struct defining the handler for the consuming Sarama method
+type consumerGroupHandler struct {
+}
+
+func (cgh *consumerGroupHandler) Setup(sarama.ConsumerGroupSession) error {
+	log.Printf("Consumer group handler setup\n")
```

Comment:
Given Peter's feedback about noise levels (and just a general eye towards growing the project), I'm wondering if long-term we shouldn't consider a logging framework to allow some differentiation between log levels. Not a show-stopper for the PR, but we should probably capture that somewhere to revisit. I can add a JIRA if you agree.

Reply:
Totally agree. We need something better than just log. As already mentioned just a PR to start up something working.